hw/system/occamy/WALKTHROUGH.md

```diff
@@ -0,0 +1,406 @@
+cd /usr/scratch/[your machine]/[your username]
+# You can find the name of the machine by running
+hostname
+# (Note, keep only the name before .ee.ethz.ch)
+```
+
+### Dependencies
+
+At IIS the default version of some tools (`gcc`, `cmake`, ...) might be too old for certain projects. You will need to setup your own default binary for these tools:
+
+```bash
+# Create your own bin folder in your home directory
+mkdir ~/bin && cd ~/bin
+# There you can change the default binaries for your user
+ln -s /usr/pack/gcc-9.2.0-af/linux-x64/bin/gcc gcc
+ln -s /usr/pack/gcc-9.2.0-af/linux-x64/bin/g++ g++
+ln -s /usr/sepp/bin/cmake-3.18.1 cmake
+ln -s /home/colluca/bin/spike-dasm spike-dasm
+# Now you need to add this folder to your PATH:
+# Open ~/.profile and add the lines
+export PATH=~/bin:$PATH
+export PATH=/usr/scratch/dachstein/colluca/opt/verible/bin:$PATH
+```
+
+Create a Python virtual environment:
+
+```
+python3.9 -m venv ~/.venvs/snitch
+source ~/.venvs/snitch/bin/activate
+```
+
+Add the last line to your `~/.profile` if you want the virtual environment to be activated by default when you open a new terminal.
+
+## Cloning and compiling Occamy
+
+First, clone this repository on your scratch folder. We suggest you first make a private fork of the repo.
+
+```bash
+git clone https://github.com/pulp-platform/snitch.git --branch=occamy-sw
+```
+
+Now install the required Python dependencies. Make sure you have activated your virtual environment before doing so.
+
+```
+pip install -r python-requirements.txt
+```
+
+Go to the occamy folder, where most of your efforts will take place:
+
+```
+cd hw/system/occamy
+```
+
+___Note:__ from now on, assume all the path to be relative to `hw/system/occamy`._
+
+First, the default configuration of Occamy is too large for a fast RTL simulation. Therefore, open `./src/occamy_cfg.hjson`, and reduce `nr_s1_quadrant` and `nr_clusters` (e.g. both to `1`). To make the changes effective, run the following command:
+
+```bash
+make update-rtl -B
+```
+
+Then compile the hardware:
+```bash
+# Compile the RTL for Questasim
+make bin/occamy_top.vsim
+```
+
+This `make` target compiled the RTL simulation model in `./work-vsim` and the [frontend server (fesvr)](https://github.com/riscv-software-src/riscv-isa-sim) C++ sources into `./work`. It also generated a script `./bin/occamy_top.vsim` (_you can read it_) that you can use to start a Questasim session initialized with the ELF of the app/kernel you want to run.
+This script relies on the `fesvr` utilities to connect to the RTL simulation and load your ELF program into the simulated DRAM memory.
+
+You can now setup your toolchain. There are plenty of pre-compiled RISC-V toolchains at IIS, for Snitch you should use the following GCC toolchain.
+
+```bash
+# You can add this to your ~/.profile such that you do not have to run this command every time you open a new terminal
+export PATH=/home/colluca/workspace/riscv/bin/:$PATH
```

Comment:
we have already a system-wide installation for the correct toolochain: 
`export PATH=/usr/pack/riscv-1.0-kgf/pulp-llvm-0.12.0/bin/:$PATH`